src/rp2_common/pico_stdio_usb/stdio_usb_descriptors.c

```diff
@@ -121,6 +124,18 @@ const uint16_t *tud_descriptor_string_cb(uint8_t index, uint16_t langid) {
     #define DESC_STR_MAX (20)
     static uint16_t desc_str[DESC_STR_MAX];
 
+    // lazy update the serial number if not set yet
+    if (!usbd_serial_str[0]) {
+        pico_unique_board_id_t id;
+        pico_get_unique_board_id(&id);
+
+        for (int i = 0; i < PICO_UNIQUE_BOARD_ID_SIZE_BYTES; i++) {
```

Comment:
yeah, we should pull in printf etc. just to have the serial number. please replace with a trivial loop to fill in the hex values directly